code/__HELPERS/priority_announce.dm

```diff
@@ -3,8 +3,10 @@
 		return
 
 	var/announcement
-	if(!istype(sound, /sound))
-		sound = SSstation.announcer.event_sounds[sound] || SSstation.announcer.get_rand_alert_sound()
+	if(!sound)
+		sound = SSstation.announcer.get_rand_alert_sound()
+	else if(SSstation.announcer.event_sounds[sound])
+		sound = SSstation.announcer.event_sounds[sound]
```

Comment:
while you're at it, you might want to make sure that these sounds are, at some point, wrapped in `sound('sound_go_here.ogg'`, otherwise, they'll inherit the echo of whatever was the previous sound the client heard.

Reply:
oh, the sound gets wrapped before it gets sent at the bottom of this proc, already

Reply:
ahh, I missed it: `var/s = sound(sound)`